lstchain/high_level/interpolate.py

```diff
@@ -0,0 +1,577 @@
+import numpy as np
+import logging
+import astropy.units as u
+from astropy.table import QTable, Table
+from astropy.io import fits
+from astropy.time import Time
+
+from lstchain.__init__ import __version__
+
+from pyirf.io.gadf import (
+    create_aeff2d_hdu,
+    create_energy_dispersion_hdu,
+    create_psf_table_hdu,
+)
+from pyirf.interpolation import (
+    interpolate_effective_area_per_energy_and_fov,
+    interpolate_energy_dispersion,
+    interpolate_psf_table,
+    # interpolate_rad_max,
+)
+from scipy.spatial import Delaunay, distance
+from scipy.interpolate import griddata
+
+log = logging.getLogger(__name__)
+
+
+def interp_params(params_list, data):
+    """
+    From a given list of angular parameters, to be used for interpolation,
+    take values from a given data table/dict.
+
+    Returns the neccessary values with applied functions as need be for the
+    interpolation, for each parameter as a list.
+    """
+    mc_pars = []
+    if "ZEN_PNT" in params_list:
+        mc_pars.append(np.cos(u.Quantity(data["ZEN_PNT"], "deg").to_value(u.rad)))
+
+    if "B_DELTA" in params_list:
+        mc_pars.append(np.sin(u.Quantity(data["B_DELTA"], "deg").to_value(u.rad)))
+
+    if "AZ_PNT" in params_list:
+        mc_pars.append(u.Quantity(data["AZ_PNT"], "deg").to_value(u.rad))
+
+    return mc_pars
+
+
+def check_in_delaunay_triangle(irfs, data_params, use_nearest_irf_node=False):
+    """
+    From a given list of IRFs as grid points used for interpolation, retrieve
+    the Delaunay triangulation list of IRFs, where the simplex includes the
+    target points in data_params.
+
+    If the target point does not exist inside the simplex, the IRF
+    corresponding to the nearest grid point to the target value. This is also
+    acheived if use_nearest_irf_node is passed as True.
+
+    Fetching the nearest IRF node is also performed by checking the azimuth
+    values, as there might be more than a single node, with the same position
+    in the interpolation parameter space.
+
+    If the list of given IRFs are not enough for calculating the Delaunay
+    triangulation, an empty list is returned.
+
+    Parameters
+    ----------
+    irfs: List of IRFs to check for Delaunay triangulation.
+        'List'
+    data_pars: Dict of arrays of range of parameters of the observed data
+        in the event list, to check for interpolation.
+        'Dict'
+    use_nearest_irf_node: Boolean if we need only the nearest IRF node.
+        'Bool'
+
+    Returns
+    -------
+    irf_list: Revised list of IRFs after the checks.
+        'List'
+    """
+    # Exclude AZ_PNT as target interpolation parameter
+    d = data_params.copy()
+    d.pop("AZ_PNT", None)
+    # For the interpolation parameters only
+    data_pars_sel = [*d.keys()]
```

Comment:
I have 2 separate `dicts` because for the IRF interpolation of a single telescope, we only need 2 parameters, `ZEN_PNT` and `B_DELTA` , but we also take into account `AZ_PNT` when distinguishing between overlapping IRF nodes in the interpolation parameter space. To use `get_nearest_az_node`, I need to have the information from these 2 dicts to make the comparison. But, for the main interpolation process, we do not need `AZ_PNT` and hence the main object used for the interpolation process, is without `AZ_PNT` based information.

At the time, I could not find a better or simpler way of doing so.

Reply:
From what I understood you don't use dictionaries after. The 'd' dictionary is only used to get the keys in this block. 

This is the one I am talking about not the ones obtained with interp_params later

Reply:
Ah ok! I wanted to make some easy steps to remove the `AZ_PNT` information from the `dict` to use it further, but not "only include" `ZEN_PNT` and `B_DELTA` in a general way. The latter seemed more hard-coded that the former... I will re-check for a simpler solution.

Reply:
You are right, the current implementation is better than hard codding. And it is also good for what you do. But if you find a version without intermediate copies I would find it a bit more clear when reading the code (e.g. I had to search the code to check if d was used anywhere else)

Best answer here https://stackoverflow.com/questions/4915920/how-to-delete-an-item-in-a-list-if-it-exists 
gives informations in case we are working with a list but it is probably not adapted here.

Reply:
Thanks for the suggestion! I incorporated a simple solution.